userguide/sound_classifier/export-coreml.md

```diff
@@ -83,20 +84,63 @@ guard let audioData = try? MLMultiArray(shape:[windowSize as NSNumber],
    fatalError("Can not create MLMultiArray")
 }
 
-// Ignore any partial window at the end.
+
+let frameLength = Int64(buffer.frameLength)
+var windowNumber = 0
+var offset = 0
+var startIndex = 0
 var results = [Dictionary<String, Double>]()
-let windowNumber = wavFile.length / Int64(windowSize)
-for windowIndex in 0..<Int(windowNumber) {
-   let offset = windowIndex * windowSize
-   for i in 0...windowSize {
-      audioData[i] = NSNumber.init(value: bufferData[0][offset + i])
-   }
-   let modelInput = my_sound_classifierInput(audio: audioData)
-
-   guard let modelOutput = try? model.prediction(input: modelInput) else {
-      fatalError("Error calling predict")
-   }
-   results.append(modelOutput.categoryProbability)
+
+// Main data provider loop for items larger than 1 window
+while ((windowNumber + 1) * windowSize) <= frameLength {
+    offset = windowNumber * windowSize
+    for i in startIndex...windowSize {
+        audioData[i] = NSNumber.init(value: bufferData[0][offset + i])
+    }
+    startIndex = 1
+    
+    let modelInput = my_sound_classifierInput(audio: audioData)
+    
+    guard let modelOutput = try? model.prediction(input: modelInput) else {
+        fatalError("Error calling predict")
+    }
+    results.append(modelOutput.categoryProbability)
+    windowNumber += 1
+}
+
+if windowNumber > 0 {
+    offset = offset + windowSize
```

Comment:
I'm not really understanding why this is necessary.

Would it be cleaner to just delete this code paragraph (i.e. lines 111-113) and change line 116 to be:
`if windowNumber == 0 {`?

Reply:
glad i revisited this, much simpler now.